src/math/matrix.c

```diff
@@ -3,93 +3,98 @@
 // Copyright(c) 2022 Intel Corporation. All rights reserved.
 //
 // Author: Seppo Ingalsuo <seppo.ingalsuo@linux.intel.com>
+//         Shriram Shastry <malladi.sastry@linux.intel.com>
 
 #include <sof/math/matrix.h>
 #include <errno.h>
 #include <stdint.h>
 
 int mat_multiply(struct mat_matrix_16b *a, struct mat_matrix_16b *b, struct mat_matrix_16b *c)
 {
-	int64_t s;
-	int16_t *x;
-	int16_t *y;
-	int16_t *z = c->data;
-	int i, j, k;
-	int y_inc = b->columns;
-	const int shift_minus_one = a->fractions + b->fractions - c->fractions - 1;
+	 /* check for NULL pointers */
+	if (!a || !b || !c)
+		return -EINVAL;
 
+	/* check for dimensions compatibility */
 	if (a->columns != b->rows || a->rows != c->rows || b->columns != c->columns)
 		return -EINVAL;
 
-	/* If all data is Q0 */
-	if (shift_minus_one == -1) {
-		for (i = 0; i < a->rows; i++) {
-			for (j = 0; j < b->columns; j++) {
-				s = 0;
-				x = a->data + a->columns * i;
-				y = b->data + j;
-				for (k = 0; k < b->rows; k++) {
-					s += (int32_t)(*x) * (*y);
-					x++;
-					y += y_inc;
-				}
-				*z = (int16_t)s; /* For Q16.0 */
-				z++;
-			}
-		}
+	int64_t acc;
```

Comment:
As style comment, please keep the original style and define local variables in beginning of function.

Reply:
Done